app/models/king.rb

```diff
@@ -6,4 +6,80 @@ def valid_move?(position_2)
 			return false
 		end
 	end
+
+	def checked?(position)
+		case self.color
+		when "white"
+			self.game.pieces.where(:color => "black").find_each do |piece|
+				if piece.valid_move?(position)
+					return true
+				end
+			end
+		when "black"
+			self.game.pieces.where(:color => "white").find_each do |piece|
+				if piece.valid_move?(position)
+					return true
+				end
+			end
+		end
+		return false
+	end
+
+	def check_mate?
+		self.possible_moves.each do |move|
+			if self.valid_move?(move)
+				if self.checked?(move)
+					self.check_piece.game.find_path(self.check_piece.current_position, self.current_position).each do |step|
+						case self.color
+						when "white"
+							self.game.pieces.where(:color => "white").find_each do |piece|
+								if piece.valid_move?(step)
+									return true
+								end
+							end
+						when "black"
+							self.game.pieces.where(:color => "black").find_each do |piece|
+								if piece.valid_move?(step)
+									return true
+								end
+							end
+						end
+					end
+				end
+			end
+		end
+		return false
+	end
+
+	def possible_moves
+		moves = []
```

Comment:
I think a possible move is to not move as well, if there are other pieces on the board.